### DISCUSSION TITLE
Can visitorId change over time?

### DISCUSSION THREAD

**Discussion by ZawHtettnaing** [fingerprintjs/fingerprintjs#817] [Q&A]
Can visitor id for a browser change over time? or the visitorid is unique and never change for each browser?

**Comment by makma** [comment]
Hello @ZawHtettnaing.
Yes, they can. 
Fingerprints generated by open-source FingerprintJS might change within weeks.
VisitorIds generated by Fingerprint Pro are more stable (even for years). Nevertheless, in specific situations, they might change as well.

You can learn more about differences in [our guide](https://dev.fingerprint.com/docs/pro-vs-open-source).
